Tensile/Tests/unit/test_TensileCreateLibrary.py

```diff
@@ -175,3 +175,103 @@ def test_verifyManifest():
+
+    # Test with wrong input type
+    archs = ["gfx942:xnack-", "gfx1010", "gfx1011"]
+    with pytest.raises(AttributeError):
+        TensileCreateLibrary.parseGfxArchitectures(archs)
+
+    # Test with comma delimiter instead of semicolon delimiter
+    archs = "gfx803,gfx906,gfx941,gfx1102"
+    with pytest.raises(ValueError, match=r"(.*) is not a known Gfx architecture."):
+        TensileCreateLibrary.parseGfxArchitectures(archs)
+
+    # Test with space delimiter instead of semicolon delimiter
+    archs = "gfx900 gfx90a:xnack+ gfx1010"
+    with pytest.raises(ValueError, match=r"(.*) is not a known Gfx architecture."):
+        TensileCreateLibrary.parseGfxArchitectures(archs)
+
+    # Test with colon instead of semicolon delimiter
+    archs = "gfx900:gfx90a:xnack+:gfx1010"
+    with pytest.raises(ValueError, match=r"(.*) is not a known Gfx architecture."):
+        TensileCreateLibrary.parseGfxArchitectures(archs)
+
+    # Test with hyphen instead of semicolon delimiter
+    archs = "gfx900-gfx90a:xnack+-gfx1010"
+    with pytest.raises(ValueError, match=r"(.*) is not a known Gfx architecture."):
+        TensileCreateLibrary.parseGfxArchitectures(archs)
+
+    # Test with leading underscore delimiter
+    archs = ";gfx803;gfx906;"
+    expected = set(["gfx803", "gfx906"])
+    result = TensileCreateLibrary.parseGfxArchitectures(archs)
+    assert result == expected, f"arch `{archs}` should map to {expected} but instead maps to {result}"
+
+    # Test with leading underscore delimiter
+    archs = "_gfx803_gfx906_"
+    expected = set(["gfx803", "gfx906"])
+    result = TensileCreateLibrary.parseGfxArchitectures(archs)
+    assert result == expected, f"arch `{archs}` should map to {expected} but instead maps to {result}"
+
+    # Test an unsupported architecture - an unsupported architecture isn't
+    archs = "gfx90Z"
+    with pytest.raises(ValueError, match="`gfx90z` is not a known Gfx architecture."):
+        result = TensileCreateLibrary.parseGfxArchitectures(archs)
+
+    # Test an unsupported architecture and a supported one
+    archs = "gfx90a;gfx90Z;gfx90a"
+    with pytest.raises(ValueError, match="`gfx90z` is not a known Gfx architecture."):
+        result = TensileCreateLibrary.parseGfxArchitectures(archs)
+
+    # Test an unsupported architecture and `all` - If `all` comes second it is rejected
+    archs = "gfx90Z;all"
+    with pytest.raises(ValueError):
+        result = TensileCreateLibrary.parseGfxArchitectures(archs)
+
+    # Test an unsupported architecture and `all` - if `all` comes first it is accepted
+    archs = "all;gfx90Z"
+    expected = set(["all"])
+    result = TensileCreateLibrary.parseGfxArchitectures(archs)
+    assert result == expected, f"arch `{archs}` should map to {expected} but instead maps to {result}"
+
+def test_mapGfxArchitectures():
+    # Test `all` architectures
+    archs = set(["all"])
+    expected = set([Common.architectureMap["all"]])
```

Comment:
See comment above - probably should not set expected value using the code we are testing.